hnn_core/tests/test_network.py

```diff
@@ -809,6 +809,15 @@ def test_network_connectivity():
             kwargs[arg] = string_arg
             pick_connection(**kwargs)
 
+    # Test network with only drive information
+    # Searching for cell connection should return empty list
+    net_only_drives = Network(params, add_drives_from_params=True,
+                              legacy_mode=False)
+    indices = pick_connection(net_only_drives,
+                              src_gids='L2_basket',
+                              target_gids='L2_basket')
```

Comment:
Wouldn't it be simpler to just use an empty network:
```suggestion
    # Test network with only drive information
    # Searching for cell connection should return empty list
    empty_net = Network(params)
    indices = pick_connection(empty_net,
                              src_gids=set(empty_net.cell_types.keys()).pop(),
                              target_gids=set(empty_net.cell_types.keys()).pop())
```
You don't need to do the whole random cell type selection if you don't want, but it's nice to start moving away from relying on hardcoded cell names generally speaking.

Reply:
I'm recreating the instance in which we found this bug. It was for a Network with initialized drives but no other connectivity. When searching for a L2_basket to L2_basket it was returning evoked drives to L2_basket connections. 

https://github.com/jonescompneurolab/hnn-core/blob/98a833035ddea98236a7797365bd082a6ab24cf7/hnn_core/network.py#L279-L296

The bugged code is shown above. Following the logic for this network: 

1. First iteration searching through src should be an empty set landing at the else condition at 293. 
2. Second iteration searching through target cells should have `inner_set` non-empty. It will also land at the else condition at 293 because `conn_set` is empty. And now `inner_set` is essentially copied to `conn_set` with the union.

So the root of the issue was that the logic didn't have a stop condition to return an empty list after searching source cells.

Reply:
Oh I see, thanks for the clarification.